presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneValuesColumns.java

```diff
@@ -30,33 +30,26 @@
 import java.util.stream.Collectors;
 
 import static com.facebook.presto.sql.planner.iterative.rule.Util.pruneInputs;
+import static com.facebook.presto.util.Optionals.tryCast;
 
 public class PruneValuesColumns
         implements Rule
 {
     @Override
     public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator)
     {
-        if (!(node instanceof ProjectNode)) {
-            return Optional.empty();
-        }
-
-        ProjectNode parent = (ProjectNode) node;
-
-        PlanNode child = lookup.resolve(parent.getSource());
-        if (!(child instanceof ValuesNode)) {
-            return Optional.empty();
-        }
-
-        ValuesNode values = (ValuesNode) child;
-
-        Optional<List<Symbol>> dependencies = pruneInputs(child.getOutputSymbols(), parent.getAssignments().getExpressions());
-        if (!dependencies.isPresent()) {
-            return Optional.empty();
-        }
-
-        List<Symbol> newOutputs = dependencies.get();
+        return tryCast(node, ProjectNode.class)
+            .flatMap((ProjectNode parent) -> lookup.resolve(parent.getSource(), ValuesNode.class)
+            .flatMap((ValuesNode child) -> pruneInputs(child.getOutputSymbols(), parent.getAssignments().getExpressions())
```

Comment:
I think it's not proper because it doesn't visually explain variables visibility. Additionally, it's different than alignment in other Presto code (breaks convention). This also means that it doesn't work with automatic code formatter, which is a no-go.